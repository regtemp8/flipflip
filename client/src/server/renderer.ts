import createCache, { type EmotionCache } from '@emotion/cache'
import { configureStore } from '@reduxjs/toolkit'
import { AppStorage, SystemConstants } from 'flipflip-common'
import createAppReducer from '../store/app/slice'
import createAudioReducer from '../store/audio/slice'
import createCaptionScriptReducer from '../store/captionScript/slice'
import createClipReducer from '../store/clip/slice'
import createLibrarySourceReducer from '../store/librarySource/slice'
import createPlaylistReducer from '../store/playlist/slice'
import createSceneReducer from '../store/scene/slice'
import createSceneGroupReducer from '../store/sceneGroup/slice'
import createTagReducer from '../store/tag/slice'
import playersSlice from '../store/player/slice'
import videoClipperSlice from '../store/videoClipper/slice'
import scenePickerSlice from '../store/scenePicker/slice'
import captionScriptorSlice from '../store/captionScriptor/slice'
import sceneDetailSlice from '../store/sceneDetail/slice'
import audioLibrarySlice from '../store/audioLibrary/slice'
import sourceScraperSlice from '../store/sourceScraper/slice'
import createConstantsReducer from '../store/constants/slice'
import { fromAppStorage } from '../store/app/convert'
import createDisplayReducer from '../store/display/slice'
import createDisplayViewReducer from '../store/displayView/slice'
import createScenePlaylistItemReducer from '../store/scenePlaylistItem/slice'
import createDisplayPlaylistItemReducer from '../store/displayPlaylistItem/slice'

export function createEmotionCache(): EmotionCache {
  return createCache({ key: 'css' })
}

export function createReduxStore(
  appStorage?: AppStorage,
  constants?: SystemConstants
) {
  const state = appStorage != null ? fromAppStorage(appStorage) : undefined
  return configureStore({
    reducer: {
      app: createAppReducer(state?.app),
      audio: createAudioReducer(state?.audio),
      captionScript: createCaptionScriptReducer(state?.captionScript),
      clip: createClipReducer(state?.clip),
      librarySource: createLibrarySourceReducer(state?.librarySource),
      playlist: createPlaylistReducer(state?.playlist),
      displayPlaylistItem: createDisplayPlaylistItemReducer(
        state?.displayPlaylistItem
      ),
      scenePlaylistItem: createScenePlaylistItemReducer(
        state?.scenePlaylistItem
      ),
      scene: createSceneReducer(state?.scene),
      sceneGroup: createSceneGroupReducer(state?.sceneGroup),
      display: createDisplayReducer(state?.display),
      displayView: createDisplayViewReducer(state?.displayView),
      tag: createTagReducer(state?.tag),
      constants: createConstantsReducer(constants),
      // components
      players: playersSlice,
      videoClipper: videoClipperSlice,
      scenePicker: scenePickerSlice,
      captionScriptor: captionScriptorSlice,
      sceneDetail: sceneDetailSlice,
      audioLibrary: audioLibrarySlice,
      sourceScraper: sourceScraperSlice
    }
  })
}
