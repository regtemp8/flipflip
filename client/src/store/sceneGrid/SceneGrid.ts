import type SceneGridCell from './SceneGridCell'
import { newSceneGridCell } from './SceneGridCell'
import { copy } from 'flipflip-common'

export default interface SceneGrid {
  id: number
  name?: string
  grid: SceneGridCell[][]
}

export const initialSceneGrid: SceneGrid = {
  id: 0,
  grid: [[newSceneGridCell()]]
}

export function newSceneGrid(init?: Partial<SceneGrid>) {
  const sceneGrid = Object.assign(copy<SceneGrid>(initialSceneGrid), init)
  sceneGrid.grid = sceneGrid.grid.map((r) =>
    r.map((c) =>
      !c.sceneID ? newSceneGridCell({ sceneID: parseInt(c as any) }) : c
    )
  )
  return sceneGrid
}
