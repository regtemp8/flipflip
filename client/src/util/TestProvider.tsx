import * as React from 'react'
import { Provider } from 'react-redux'
import { createTheme } from '@mui/material'
import { StyledEngineProvider, ThemeProvider } from '@mui/material/styles'
import defaultTheme from 'flipflip-common'

export default class TestProvider extends React.Component {
  readonly props: {
    children: any
    store: any
  }

  disableTransitions = {
    defaultProps: {
      disablePortal: true,
      hideBackdrop: true,
      TransitionComponent: ({ children }) => children
    }
  }

  render() {
    const theme = createTheme({
      ...defaultTheme,
      // disable Transitions when runing tests, throws errors otherwise.
      components: {
        MuiMenu: this.disableTransitions,
        MuiPopover: this.disableTransitions,
        MuiDialog: this.disableTransitions,
        MuiModal: this.disableTransitions
      }
    } as any)
    return (
      <Provider store={this.props.store}>
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={theme}>{this.props.children}</ThemeProvider>
        </StyledEngineProvider>
      </Provider>
    )
  }
}
