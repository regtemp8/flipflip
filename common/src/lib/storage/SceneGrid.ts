import { newSceneGridCell, type SceneGridCell } from './SceneGridCell';

export type SceneGrid = {
  id: number;
  name?: string;
  grid: SceneGridCell[][];
};

export const initialSceneGrid: SceneGrid = {
  id: 0,
  grid: [[newSceneGridCell()]],
};

export function newSceneGrid(init?: Partial<SceneGrid>) {
  return {
    id: init?.id ?? initialSceneGrid.id,
    name: init?.name,
    grid: mapGrid(init?.grid) ?? initialSceneGrid.grid,
  };
}

function mapGrid(
  grid?: SceneGridCell[][] | number[][] | string[][]
): SceneGridCell[][] | undefined {
  if (grid == null) {
    return undefined;
  }

  return grid.map((r) =>
    r.map((c) => {
      const partial = c as Partial<SceneGridCell>;
      if (partial.sceneID == null) {
        const sceneID = Number.isNaN(c) ? parseInt(c as string) : (c as number);
        return newSceneGridCell({ sceneID });
      } else {
        return newSceneGridCell(partial);
      }
    })
  );
}
