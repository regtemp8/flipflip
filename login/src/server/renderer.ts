import createCache, { type EmotionCache } from "@emotion/cache";

export function createEmotionCache(nonce?: string): EmotionCache {
  return createCache({ key: "css", nonce });
}
