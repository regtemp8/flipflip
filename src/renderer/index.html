<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <title>FlipFlip</title>
    <!-- Responsive meta tag -- ensures proper rendering and touch zooming for all devices -->
    <meta name="viewport" content="minimum-scale=1, initial-scale=1, width=device-width, shrink-to-fit=no" />
    <!-- Roboto Font -->
    <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap" />
    <!-- Font Icons -->
    <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons" />

    <meta
      http-equiv="Content-Security-Policy"
      content="frame-src https://www.imagefap.com http://www.imagefap.com https://hypno.nimja.com/ http://hypno.nimja.com/; font-src 'self' https://fonts.gstatic.com; style-src 'self' https://fonts.googleapis.com 'unsafe-inline'"
    />
  </head>

  <body>
    <div id="app" class="app"></div>
  </body>
</html>
