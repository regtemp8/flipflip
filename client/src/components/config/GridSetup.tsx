import React, { type ChangeEvent, useEffect, useState } from 'react'
import clsx from 'clsx'

import {
  AppBar,
  Container,
  Fab,
  IconButton,
  TextField,
  type Theme,
  Toolbar,
  Tooltip,
  Typography
} from '@mui/material'

import createStyles from '@mui/styles/createStyles'
import withStyles, { type WithStyles } from '@mui/styles/withStyles'

import ArrowBackIcon from '@mui/icons-material/ArrowBack'
import DeleteIcon from '@mui/icons-material/Delete'
import PlayCircleOutlineIcon from '@mui/icons-material/PlayCircleOutline'

import BaseTextField from '../common/text/BaseTextField'
import { SGT, SP } from 'flipflip-common'
import {
  selectAppTutorial,
  selectAppSpecialMode
} from '../../store/app/selectors'
import {
  selectSceneGridName,
  selectSceneGridHeight,
  selectSceneGridWidth
} from '../../store/sceneGrid/selectors'
import { useAppDispatch, useAppSelector } from '../../store/hooks'
import { setSceneGridName } from '../../store/sceneGrid/actions'
import { setSceneGridCells } from '../../store/sceneGrid/slice'
import { removeSceneGrid } from '../../store/sceneGrid/thunks'
import { doneDimensionsTutorial, setRouteGoBack } from '../../store/app/thunks'
import { generateScenes } from '../../store/scene/thunks'
import { playGrid } from '../../store/player/thunks'
import GridCellSetup from './GridCellSetup'
import { convertGridIDToSceneID } from '../../data/utils'

const styles = (theme: Theme) =>
  createStyles({
    root: {
      display: 'flex',
      height: '100vh'
    },
    appBar: {
      zIndex: theme.zIndex.drawer + 1
    },
    appBarSpacer: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'flex-end',
      padding: '0 8px',
      minHeight: 64
    },
    title: {
      textAlign: 'center',
      flexGrow: 1
    },
    headerBar: {
      display: 'flex',
      alignItems: 'center',
      whiteSpace: 'nowrap',
      flexWrap: 'nowrap'
    },
    headerLeft: {
      flexBasis: '13%'
    },
    headerRight: {
      flexBasis: '13%',
      justifyContent: 'flex-end',
      display: 'flex'
    },
    titleField: {
      margin: 0,
      textAlign: 'center',
      flexGrow: 1
    },
    titleInput: {
      color: theme.palette.primary.contrastText,
      textAlign: 'center',
      fontSize: theme.typography.h4.fontSize
    },
    noTitle: {
      width: '33%',
      height: theme.spacing(7)
    },
    content: {
      display: 'flex',
      flexGrow: 1,
      flexDirection: 'column',
      backgroundColor: theme.palette.background.default
    },
    container: {
      height: '100%',
      padding: theme.spacing(0)
    },
    dimensionInput: {
      color: `${theme.palette.primary.contrastText} !important`,
      minWidth: theme.spacing(6)
    },
    grid: {
      flexGrow: 1,
      display: 'grid',
      height: '100%'
    },
    deleteButton: {
      backgroundColor: theme.palette.error.dark,
      margin: 0,
      top: 'auto',
      right: 20,
      bottom: 20,
      left: 'auto',
      position: 'fixed',
      zIndex: 3
    },
    deleteIcon: {
      color: theme.palette.error.contrastText
    },
    fill: {
      flexGrow: 1
    },
    backdropTop: {
      zIndex: theme.zIndex.modal + 1
    },
    highlight: {
      borderWidth: 2,
      borderColor: theme.palette.secondary.main,
      borderStyle: 'solid'
    },
    disable: {
      pointerEvents: 'none'
    }
  })

export interface GridSetupProps extends WithStyles<typeof styles> {
  gridID: number
}

function GridSetup(props: GridSetupProps) {
  const dispatch = useAppDispatch()
  const tutorial = useAppSelector(selectAppTutorial())
  const specialMode = useAppSelector(selectAppSpecialMode())
  const name = useAppSelector(selectSceneGridName(props.gridID))
  const height = useAppSelector(selectSceneGridHeight(props.gridID))
  const width = useAppSelector(selectSceneGridWidth(props.gridID))

  const [isEditingName, setIsEditingName] = useState(false)
  const [isEditing, setIsEditing] = useState([-1, -1])

  if (tutorial === SGT.dimensions && width === 2 && height === 2) {
    dispatch(doneDimensionsTutorial(props.gridID))
  }

  useEffect(() => {
    setIsEditingName(specialMode === SP.autoEdit)
  }, [specialMode])

  const onUpdateSceneGridCells = (height: number, width: number) => {
    dispatch(setSceneGridCells({ id: props.gridID, row: height, col: width }))
    if (tutorial === SGT.cells && width === 2 && height === 2) {
      dispatch(doneDimensionsTutorial(props.gridID))
    }
  }

  const onUpdateHeight = (height: number) => {
    onUpdateSceneGridCells(height, width)
  }

  const onUpdateWidth = (width: number) => {
    onUpdateSceneGridCells(height, width)
  }

  const onChangeNumber =
    (min: number, max: number, action: (value: number) => void) =>
    (event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
      let value = event.target.value !== '' ? Number(event.target.value) : min
      value = Math.max(value, min)
      value = Math.min(value, max)
      action(value)
    }

  const onChangeHeight = onChangeNumber(1, 5, onUpdateHeight)
  const onChangeWidth = onChangeNumber(1, 5, onUpdateWidth)

  const startEditingGridCell = (row: number, col: number) => {
    console.log(`EDIT ${row}, ${col}`)
    setIsEditing([row, col])
  }

  const endEditingGridCell = (row: number, col: number) => {
    if (isEditing[0] === row && isEditing[1] === col) {
      setIsEditing([-1, -1])
    }
  }

  const beginEditingName = () => {
    setIsEditingName(true)
  }

  const endEditingName = () => {
    setIsEditingName(false)
  }

  const goBack = () => {
    dispatch(setRouteGoBack())
  }

  const onPlayGrid = () => {
    // Regenerate scene(s) before playback
    dispatch(generateScenes(convertGridIDToSceneID(props.gridID)))
    dispatch(playGrid(props.gridID))
  }

  const classes = props.classes
  const colSize = 100 / width
  const rowSize = 100 / height
  let gridTemplateColumns = ''
  let gridTemplateRows = ''
  for (let w = 0; w < width; w++) {
    gridTemplateColumns += colSize.toString() + '% '
  }
  for (let h = 0; h < height; h++) {
    gridTemplateRows += rowSize.toString() + '% '
  }

  return (
    <div className={classes.root}>
      <AppBar
        enableColorOnDark
        position="absolute"
        className={clsx(
          classes.appBar,
          tutorial === SGT.dimensions && classes.backdropTop
        )}
      >
        <Toolbar className={classes.headerBar}>
          <div className={classes.headerLeft}>
            <Tooltip disableInteractive title="Back" placement="right-end">
              <IconButton
                edge="start"
                color="inherit"
                aria-label="Back"
                className={clsx(tutorial === SGT.dimensions && classes.disable)}
                onClick={goBack}
                size="large"
              >
                <ArrowBackIcon />
              </IconButton>
            </Tooltip>
          </div>

          {isEditingName && (
            <form onSubmit={endEditingName} className={classes.titleField}>
              <BaseTextField
                variant="standard"
                autoFocus
                fullWidth
                id="title"
                margin="none"
                inputProps={{ className: classes.titleInput }}
                selector={selectSceneGridName(props.gridID)}
                action={setSceneGridName(props.gridID)}
                onBlur={endEditingName}
              />
            </form>
          )}
          {!isEditingName && (
            <Typography
              component="h1"
              variant="h4"
              noWrap
              className={clsx(
                classes.title,
                name.length === 0 && classes.noTitle,
                tutorial === SGT.dimensions && classes.disable
              )}
              onClick={beginEditingName}
            >
              {name}
            </Typography>
          )}

          <div className={classes.headerRight}>
            <TextField
              label="Height"
              margin="dense"
              value={height}
              onChange={onChangeHeight}
              variant="filled"
              className={clsx(tutorial === SGT.dimensions && classes.highlight)}
              InputLabelProps={{ className: classes.dimensionInput }}
              inputProps={{
                className: classes.dimensionInput,
                min: 1,
                max: 5,
                type: 'number'
              }}
            />
            <TextField
              label="Width"
              margin="dense"
              value={width}
              onChange={onChangeWidth}
              variant="filled"
              className={clsx(tutorial === SGT.dimensions && classes.highlight)}
              InputLabelProps={{ className: classes.dimensionInput }}
              inputProps={{
                className: classes.dimensionInput,
                min: 1,
                max: 5,
                type: 'number'
              }}
            />
            <IconButton
              edge="end"
              color="inherit"
              aria-label="Play"
              className={clsx(tutorial === SGT.dimensions && classes.disable)}
              onClick={onPlayGrid}
              size="large"
            >
              <PlayCircleOutlineIcon fontSize="large" />
            </IconButton>
          </div>
        </Toolbar>
      </AppBar>

      <main className={classes.content}>
        <div className={classes.appBarSpacer} />
        <Container maxWidth={false} className={classes.container}>
          <div
            className={classes.grid}
            style={{
              gridTemplateColumns,
              gridTemplateRows
            }}
          >
            {[...Array(height).keys()].map((row) => (
              <React.Fragment key={row}>
                {[...Array(width).keys()].map((col) => (
                  <GridCellSetup
                    key={row + '-' + col}
                    id={props.gridID}
                    row={row}
                    col={col}
                    isEditing={isEditing[0] === row && isEditing[1] === col}
                    onStartEditing={startEditingGridCell}
                    onEndEditing={endEditingGridCell}
                  />
                ))}
              </React.Fragment>
            ))}
          </div>
          <Fab
            className={classes.deleteButton}
            onClick={() => {
              dispatch(removeSceneGrid(props.gridID))
            }}
            size="small"
          >
            <DeleteIcon className={classes.deleteIcon} />
          </Fab>
        </Container>
      </main>
    </div>
  )
}

;(GridSetup as any).displayName = 'GridSetup'
export default withStyles(styles)(GridSetup as any)
