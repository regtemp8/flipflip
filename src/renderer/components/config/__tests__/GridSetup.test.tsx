import React from "react";
import { describe, it, expect } from "@jest/globals";
import renderer from "react-test-renderer";
import GridSetup from "../GridSetup";
import TestProvider from "../../../../../test/util/TestProvider";
import SceneGrid from "../../../data/SceneGrid";

jest.mock('../../configGroups/SceneSelect', () => 'SceneSelect');

// mocking this so that test doesn't throw error
jest.mock('@mui/material/MenuList', () => 'MenuList');

describe("GridSetup", () => {
  it("should match snapshot", () => {
    const sceneGrid = new SceneGrid({ name: 'test' })
    const component = renderer.create(
      <TestProvider>
        <GridSetup
            allScenes={[]}
            autoEdit={false}
            scene={sceneGrid}
            tutorial={null}
            goBack={() => {}}
            onDelete={(grid) => {}}
            onGenerate={(scene, children, force) => {}}
            onPlayGrid={(grid) => {}}
            onTutorial={(tutorial) => {}}
            onUpdateGrid={(grid, fn) => {}}
        />
      </TestProvider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
