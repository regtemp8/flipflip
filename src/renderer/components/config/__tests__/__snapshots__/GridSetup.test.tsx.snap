// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridSetup should match snapshot 1`] = `
<div
  className="GridSetup-root-1"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionAbsolute GridSetup-appBar-2 css-iu2t21-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular GridSetup-headerBar-5 css-hyum1k-MuiToolbar-root"
    >
      <div
        className="GridSetup-headerLeft-6"
      >
        <button
          aria-label="Back"
          aria-labelledby={null}
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-184mzhy-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element={true}
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowBackIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <h1
        className="MuiTypography-root MuiTypography-h4 MuiTypography-noWrap GridSetup-title-4 css-x00mp-MuiTypography-root"
        onClick={[Function]}
      >
        test
      </h1>
      <div
        className="GridSetup-headerRight-7"
      >
        <div
          className="MuiFormControl-root MuiFormControl-marginDense MuiTextField-root css-4mzek5-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled GridSetup-dimensionInput-13 css-104bnou-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={true}
            htmlFor="mui-2"
            id="mui-2-label"
          >
            Height
          </label>
          <div
            className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-dtbfm4-MuiInputBase-root-MuiFilledInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiFilledInput-input GridSetup-dimensionInput-13 css-10botns-MuiInputBase-input-MuiFilledInput-input"
              disabled={false}
              id="mui-2"
              max={5}
              min={1}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="number"
              value={1}
            />
          </div>
        </div>
        <div
          className="MuiFormControl-root MuiFormControl-marginDense MuiTextField-root css-4mzek5-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled GridSetup-dimensionInput-13 css-104bnou-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={true}
            htmlFor="mui-3"
            id="mui-3-label"
          >
            Width
          </label>
          <div
            className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-dtbfm4-MuiInputBase-root-MuiFilledInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiFilledInput-input GridSetup-dimensionInput-13 css-10botns-MuiInputBase-input-MuiFilledInput-input"
              disabled={false}
              id="mui-3"
              max={5}
              min={1}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="number"
              value={1}
            />
          </div>
        </div>
        <button
          aria-label="Play"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeEnd MuiIconButton-sizeLarge css-1f5y6oa-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
            data-testid="PlayCircleOutlineIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m10 16.5 6-4.5-6-4.5v9zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
            />
          </svg>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
  <main
    className="GridSetup-content-11"
  >
    <div
      className="GridSetup-appBarSpacer-3"
    />
    <div
      className="MuiContainer-root GridSetup-container-12 css-ur2jdm-MuiContainer-root"
    >
      <div
        className="GridSetup-grid-14"
        style={
          {
            "gridTemplateColumns": "100% ",
            "gridTemplateRows": "100% ",
          }
        }
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium GridSetup-gridCell-15 react-draggable css-1g0jmg2-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          id="0-0"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            {
              "transform": "translate(0px,0px)",
            }
          }
          tabIndex={0}
          transform={null}
          type="button"
        >
          
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        TransitionComponent={[Function]}
        className="MuiPopover-root MuiMenu-root MuiModal-root MuiModal-hidden css-6f6ff7-MuiModal-root-MuiPopover-root-MuiMenu-root"
        id="scene-menu"
        onKeyDown={[Function]}
        role="presentation"
      >
        <div
          data-testid="sentinelStart"
          onFocus={[Function]}
          tabIndex={-1}
        />
        <div
          className="MuiPaper-root MuiMenu-paper GridSetup-sceneMenu-16 MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiPopover-paper css-1yuf3k2-MuiPaper-root-MuiPopover-paper-MuiMenu-paper"
          style={
            {
              "opacity": 0,
            }
          }
        >
          <MenuList
            actions={
              {
                "current": null,
              }
            }
            autoFocus={true}
            autoFocusItem={false}
            className="MuiMenu-list css-rpp6kk-MuiMenu-list"
            onKeyDown={[Function]}
            variant="selectedMenu"
          />
        </div>
        <div
          data-testid="sentinelEnd"
          onFocus={[Function]}
          tabIndex={-1}
        />
      </div>
      <button
        className="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default GridSetup-deleteButton-17 css-12m9wit-MuiButtonBase-root-MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium GridSetup-deleteIcon-18 css-i4bv87-MuiSvgIcon-root"
          data-testid="DeleteIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </main>
</div>
`;
